torch/_inductor/index_propagation.py

```diff
@@ -256,7 +257,11 @@ def indirect_indexing(
 
         # indirect_indexing returns a sympy value, so no need to wrap in IndexPropVar here
         if isinstance(index, IndexPropVar) and index.is_symbolic:
-            # If we are turning a indirect indexing into direct, we need to wrap it.
-            index = index.value.expr
-            return index + Where(index >= 0, 0, size)
+            bounds = bound_sympy(index.value.expr)
+            # Code below imposes a regression on PyTorch code as only bounded
+            # variables are allowed to skip a TORCH_CHECK for getitem
```

Comment:
Can we add a new operation `ops.check_bounds(expr, size)` that checks the value is withing bounds without reverting back to indirect indexing here? The regression here is much more significant than just adding an assert as indirect indexing can block potential fusions.